src/errors.rs

```diff
@@ -22,6 +22,16 @@ impl StdError for BodyError {
     fn description(&self) -> &str {
         &self.detail[..]
     }
+
+    fn cause(&self) -> Option<&StdError> {
+        use BodyErrorCause::*;
+
+        match self.cause {
```

Comment:
I would have expected it to infer that from the typesig, and auto-check that all variants impl that trait.